mclf/semistable_reduction/admissible_reduction.py

```diff
@@ -45,19 +45,12 @@
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
 
-from sage.structure.sage_object import SageObject
-from sage.rings.polynomial.polynomial_ring_constructor import PolynomialRing
-from sage.rings.function_field.constructor import FunctionField
-from sage.misc.cachefunc import cached_method
-from sage.rings.infinity import Infinity
-from sage.functions.other import floor
+from sage.all import SageObject, PolynomialRing, FunctionField, cached_method, Infinity, floor
```

Comment:
@swewers: On readthedocs.org there is no sage avalable. All the imports have to go through the "fake" Sage that is configured at the bottom of `docs/conf.py`. If you need to import something that is not in `sage.all` you need to add it there.

Reply:
The "MagicMock" used there does not do anything useful but tries to never throw an error no matter what you do to it:
```
>>> from unittest.mock import MagicMock
>>> X = MagicMock()
>>> X.y ** X.z()
<MagicMock name='mock.y.__pow__()' id='140534834155080'>
```
That's usually good enough for docbuilding. The things don't need to work, they just need to import.

Reply:
Does that mean that docs/conf.py also has to knwo about the relative import within mclf? 
At the moment, http://mclf.readthedocs.io/en/latest/smooth_projective_curves.html shows some of the functions, but not all. Similar for the other modules.